src/FFmpegReader.cpp

```diff
@@ -991,44 +978,51 @@ std::shared_ptr<Frame> FFmpegReader::ReadStream(int64_t requested_frame) {
 		// if the has_video or has_audio properties are manually overridden)
 		if ((!info.has_video && packet && packet->stream_index == videoStream) ||
 			(!info.has_audio && packet && packet->stream_index == audioStream)) {
+			// Keep track of deleted packet counts
+			if (packet->stream_index == videoStream) {
+				packet_status.video_decoded++;
+			} else if (packet->stream_index == audioStream) {
+				packet_status.audio_decoded++;
+			}
+
+			// Remove unused packets (sometimes we purposely ignore video or audio packets,
+			// if the has_video or has_audio properties are manually overridden)
 			RemoveAVPacket(packet);
 			packet = NULL;
-			packets_decoded++;
 		}
 
 		// Determine end-of-stream (waiting until final decoder threads finish)
 		// Force end-of-stream in some situations
-		end_of_file = packets_eof && video_eof && audio_eof;
-		if ((packets_eof && packets_read == packets_decoded) || end_of_file) {
+		packet_status.end_of_file = packet_status.packets_eof && packet_status.video_eof && packet_status.audio_eof;
+		if ((packet_status.packets_eof && packet_status.packets_read() == packet_status.packets_decoded()) || packet_status.end_of_file) {
```

Comment:
Looks like this should always break, since we decode as long as there are packets read.